org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/launch/npm/NpmLaunchDelegate.java

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.wildwebdeveloper.launch.npm;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.model.ILaunchConfigurationDelegate;
+import org.eclipse.jface.dialogs.ErrorDialog;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.wildwebdeveloper.Activator;
+import org.eclipse.wildwebdeveloper.InitializeLaunchConfigurations;
+import org.eclipse.wildwebdeveloper.debug.AbstractHTMLDebugDelegate;
+
+public class NpmLaunchDelegate implements ILaunchConfigurationDelegate {
+
+	static final String ID = "org.eclipse.wildwebdeveloper.launchConfiguration.NPMLaunch";
+
+	@Override
+	public void launch(ILaunchConfiguration configuration, String mode, ILaunch launch, IProgressMonitor monitor)
+			throws CoreException {
+
+		File packageJSONDirectory = new File(
+				configuration.getAttribute(AbstractHTMLDebugDelegate.CWD, "No package.json directory path set").trim());
+		File packageJSON = new File(
+				configuration.getAttribute(AbstractHTMLDebugDelegate.PROGRAM, "No package.json path set").trim());
+		String argumentString = configuration.getAttribute(AbstractHTMLDebugDelegate.ARGUMENTS, "No NPM argument set")
+				.trim();
+		List<String> arguments = new ArrayList<>();
+		arguments.add(findNPMLocation());
+		arguments.addAll(Arrays.asList(argumentString.split(" ")));
```

Comment:
Maybe use a better regex in a future iteration? Something that will account for quotes (ie `npm run echo "hello world"`)